### ISSUE TITLE
Native Image Generation Failure | NoSuchFileException | Windows 10

### ISSUE THREAD

**Issue by harpreetsingh29** [quarkusio/quarkus-quickstarts#407]
Error while building native image on getting-started in windows 10 (pom has 1.0.1 FInal and even tested with the next version):
```
Error: Invalid Path entry getting-started-1.0-SNAPSHOT-runner.jar
java.nio.file.NoSuchFileException: /project/getting-started-1.0-SNAPSHOT-runner.jar
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.0.1.Final:native-image (default) on project getting-started: Failed to generate a native image: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:289)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:497)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:941)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
[ERROR]         at io.quarkus.builder.BuildContext$$Lambda$95/1029023736.run(Unknown Source)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR]         at java.lang.Thread.run(Thread.java:745)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.lang.RuntimeException: Image generation failed
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:278)
[ERROR]         ... 13 more
```
[FullLogs.txt](https://github.com/quarkusio/quarkus-quickstarts/files/3982012/FullLogs.txt)

Note: I only have C Drive in my system.

**System/Software Configuration Details**:
![image](https://user-images.githubusercontent.com/28382120/71148688-5495fd80-2252-11ea-959a-d23412be401f.png)

![image](https://user-images.githubusercontent.com/28382120/71148712-61b2ec80-2252-11ea-886f-5f2ee9b63443.png)

![image](https://user-images.githubusercontent.com/28382120/71148980-3381dc80-2253-11ea-9cdd-b7518fd5e016.png)

![image](https://user-images.githubusercontent.com/28382120/71148992-3d0b4480-2253-11ea-8094-d62178dfe2eb.png)

**Comment by laudukang** [comment]
```
./mvnw package -Pnative -Dquarkus.native.container-runtime=docker
```
same issue in win10 1909, docker desktop same as above
quarkus.platform.version: 1.1.1.Final

**Comment by BogdanSukonnov** [comment]
I had the same issue. Solved it by sharing C drive in Docker settings. I think it needs to be in guide.
![image](https://user-images.githubusercontent.com/49834667/80314667-8ff59980-87fb-11ea-9ab3-cbf44baa7f1c.png)

**Comment by opeti** [comment]
Same issue here. Windows 10 Pro and Docker dekstop 2.2.0.5
My drive is shared with Docker. The problem is that the plugin tries to mount it the following way:
`docker run -v //d/work/bar/target/bar-1.0.0-SNAPSHOT-native-image-source-jar:/project:z ...`
This form is not mounting the specified path to the Docker container. If I manually replace the path in the command to Windows format: `docker run -v d:\work\bar\target\bar-1.0.0-SNAPSHOT-native-image-source-jar:/project:z` then it is working fine.

**Comment by cescoffier** [comment]
I don't have a windows machine to check. If I'm not mistaken @maxandersen have one. If the changes work, we should fix the command for windows.

**Comment by opeti** [comment]
Strange. I don't know what happened, but now the volume mounting works fine also with double forward slashes (//d/work... form in my previous comment), so the native image generation is also successful when running it from Maven.
Actually I can't reproduce the issue I mentioned yesterday. Unfortunately I haven't found in Docker documentation which is the preferred form in Windows OS. Actually it works both ways, so probably it was some Docker issue in my case.

**Comment by cescoffier** [comment]
Closing - see last comment.

**Comment by maxandersen** [comment]
@opeti any chance you ran these tests in different terminal/shells on windows ? I've seen cases where cygwin vs command.exe vs windows new terminal path resolution worked slightly different for various arcane reasons.

**Comment by opeti** [comment]
@maxandersen Nope, I've run all my tests using cmd.exe.

But I figured out what the problem was. Long story short: It is a Docker Desktop glitch, nothing to do with Quarkus. We just have to share our drive with Docker and everything should work fine.

Detailed description:
In Docker Desktop for Windows it is necessary to select which drives are shared  with Docker for mounting. When I've tried it for the first time, no drive was shared with Docker. In earlier versions of Docker Desktop I've got an error message when I tried to mount a folder from a non shared drive. In the actual version of Docker Desktop a Windows notification appears instead of an error where I can share the drive with one click. But this notification appears only if I use Windows style path for mounting (`d:\...` and not `//d/...`) If my drive is not shared and I mount it with `-v //d/...` option, then I don't get error message nor notification, the container starts, but nothing is mounted.
A Docker Desktop has a little strange behavior after sharing a drive. If I share a drive (in settings or by clicking on the notification) then mounting works inmediately with Windows style paths, but it is still not working with the `//d/...` format (no error message, just nothing gets mounted). However after restarting Docker Desktop both formats are working fine. I've tried it several times by unsharing the drive, sharing it again, and this behavior is consistent.

So the summary:
1. If the drive is not marked for sharing, then it can not be mounted.
2. After sharing the drive it can be mounted with Windows path format.
3. After restarting Docker Desktop (or the whole system) mounting works both with Windows and Unix path format.

The Quarkus plugin specifies the volume mount in Unix path format, so it works only if the drive was previously shared. In other case we must share it and restart Docker Desktop to succeed.
